### ISSUE TITLE
Strings in bindings are not allowed to contain interpolation syntax

### ISSUE THREAD

**Issue by JoostK** [angular/angular#39601]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/compiler

### Is this a regression?
Nope

### Description

A parse error occurs when interpolation syntax is used inside of a string literal in a binding: e.g. `<comp [input]="'{{interpolation}}'"></comp>` is valid but is rejected with an error:

```
Parser Error: Got interpolation ({{}}) where expression was expected at column 1 in ['{{interpolation}}'] in /~/src/app/app.component.html@0:13
```

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-ivy-wjcqqs?file=src%2Fapp%2Fapp.component.html

## 🌍  Your Environment

**Angular Version:**
Tested on 10.2.2 but has always been broken.

**Comment by ttma1046** [comment]
why not `[input]="interpolation"`?
with
in the ts
```
this.interpolation = `${whichyouwant}`
```

**Comment by basherr** [comment]
@ttma1046 the idea is to pass a string interpolation to child component but rather to evaulate

**Comment by JoostK** [comment]
@ttma1046 This issue was created to capture a defect in the parser behavior, it's just a hypothetical example.

**Comment by JoostK** [comment]
Reopening this as this wasn't actually addressed in dc6d40e5bcf7d4dab61776b94a83024a0943f3b3.
